src/resolve/resolved-manager.c

```diff
@@ -62,6 +65,23 @@ static int manager_process_link(sd_netlink *rtnl, sd_netlink_message *mm, void *
         if (r < 0)
                 goto fail;
 
+        if (!set_isempty(m->restrict_interfaces)) {
+                char ifname[IF_NAMESIZE + 1];
+                bool found = false;
+
+                if (format_ifname_full(ifindex, ifname, FORMAT_IFNAME_IFINDEX)) {
+                        SET_FOREACH(p, m->restrict_interfaces, j)
+                                if (!fnmatch(p,  ifname, 0)) {
+                                        found = true;
+                                        break;
+                                }
+                }
+
+                if (!found)
+                        return 0;
+
+        }
```

Comment:
Maybe, the check should be split out as a new function, then we can reduce the indentation.